scripts/integration-test

```diff
@@ -1,83 +1,55 @@
-TAIL_PID=$!
+# Wait for rancher to start up
+try --delay 2 --max 30 --waitmsg "Waiting for rancher/webhook to be deployed" --failmsg "No rancher here" kubectl rollout status --watch=true --timeout=10s -n cattle-system deploy/rancher-webhook
+echo "Rancher deployed"
+
+webhook_deployed() {
+    status=$(kubectl get apps.catalog.cattle.io -n cattle-system rancher-webhook -o jsonpath="{@.status.summary.state}") && [[ "$status" == "deployed" ]]
+}
 
 # Wait for Rancher to deploy rancher-webhook.
-while ! kubectl rollout status -w -n cattle-system deploy/rancher-webhook >/dev/null 2>&1; do
-    echo "Waiting for rancher to deploy rancher-webhook..."
-    sleep 2
-done
+try --delay 2 --max 30 --waitmsg "Waiting for webhook to be deployed (2)" webhook_deployed
 echo "Webhook deployed"
 
-# After rancher deploys webhook kill the bash command running tail.
-kill ${TAIL_PID}
-
 # Wait for helm operation to complete and save rancher-webhook release info before we kill rancher and the cluster.
-while 
-	status=$(kubectl get apps.catalog.cattle.io -n cattle-system rancher-webhook -o jsonpath="{@.status.summary.state}")
-	[[ "$status" != "deployed" ]]
-do 
-	echo "Waiting for helm operation to finish, current status $status"
-	sleep 2
-done
-
-# Kill Rancher since we only need the CRDs and the initial webhook values.
-# We do not want Rancher to reconcile an older version of the webhook during test.
-kill ${RANCHER_PID}
+app_catalog_deployed() {
+  status=$(kubectl get apps.catalog.cattle.io -n cattle-system rancher-webhook -o jsonpath="{@.status.summary.state}") && [[ "$status" == "deployed" ]]
+}
 
-echo "Rancher has been stopped starting K3s."
-# Start Cluster without Rancher.
-k3s server --cluster-init --disable=traefik,servicelb,metrics-server,local-storage --node-name=local-node --log=./k3s.log >/dev/null 2>&1 &
-KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+try --delay 2 --max 30 --waitmsg "Waiting for app-catalog to be helm-deployed (2)" app_catalog_deployed
+
+# Shut down the core rancher part, but leave the rest of the rancher environment running
 
-# Wait for cluster to start.
-while ! kubectl version >/dev/null 2>&1; do
-    echo "Waiting for cluster to start"
-    sleep 5
-done
+kubectl scale deploy rancher -n cattle-system --replicas=0
+kubectl wait pods -l app=rancher-webhook --for=delete --namespace cattle-system
+
+echo "Rancher has been stopped starting K3s."
 
 echo "Uploading new webhook image"
 
-###### Upload the newly created webhook image to containerd, then install the webhook chart using the new image
-IMAGE_FILE=./dist/rancher-webhook-image.tar
-# import image to containerd and get the image name
-WEBHOOK_REPO=$(ctr image import ${IMAGE_FILE} | cut -d ' ' -f 2 | cut -d ':' -f 1)
+#### NO!!! Use ghcr.io ###### Upload the newly created webhook image to containerd, then install the webhook chart using the new image
+#### IMAGE_FILE=./dist/rancher-webhook-image.tar
+#### k3d image import $IMAGE_FILE -c ${CLUSTER_NAME}
```

Comment:
Are these comments still up to date?

Reply:
Cruft for while I was trying to get the image into `k3d`